### PR TITLE
Update documentation and behaviour in SLIC

### PR DESCRIPTION
**Pull request by jni**
- add reference to SLIC0
- `convert2lab` defaults to `None`, which is equivalent to `True` except that it only works when the last channel is length 3. When it is not, `None` just works, while `True` raises a `ValueError`.

Annoyingly, I encountered this sporadically during my testing:

```
======================================================================
FAIL: skimage.restoration.tests.test_unwrap.test_wrap_around(3, 1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/nuneziglesiasj/anaconda/envs/skimtest/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "skimage/restoration/tests/test_unwrap.py", line 106, in check_wrap_around
    image_unwrap_wrap_around[index_last])
  File "/Users/nuneziglesiasj/anaconda/envs/skimtest/lib/python2.7/site-packages/numpy/testing/utils.py", line 470, in assert_almost_equal
    raise AssertionError(msg)
AssertionError: 
Arrays are not almost equal to 7 decimals
 ACTUAL: -25.132741228718345
 DESIRED: 12.566370614359172
```

=(

### CODE REVIEW

**Comment by stefanv** [comment]
@jni Are you running the latest build?  The phase unwrapping code should now be fully deterministic.

**Comment by jni** [comment]
@stefanv I know that. Yes, I branched off the latest master. Hence the sad face. =( I guess a wtf face might be more appropriate!

File: skimage/segmentation/slic_superpixels.py [hunk 1]
```diff
@@ -58,7 +59,7 @@ def slic(image, n_segments=100, compactness=10., max_iter=10, sigma=0,
         Proportion of the maximum connected segment size. A value of 3 works
         in most of the cases.
     slic_zero: bool, optional
-        Run SLIC-zero, the zero-parameter mode of SLIC
+        Run SLIC-zero, the zero-parameter mode of SLIC. [2]
```
Comment (stefanv) [inline skimage/segmentation/slic_superpixels.py]: ``[2]_``

**Comment by stefanv** [comment]
Thank you!

**Pull request merged**